### PR TITLE
Docker registry

### PR DESCRIPTION
**Pull request by tombentley**
### Type of change

- Bugfix
- ~Enhancement / new feature~
- ~Refactoring~

### Description

This PR always pushed docker images to a local registry, which means the images used for the system test are the one produced by the build. If the system tests pass the images are pushed to docker hub.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] ~Write tests~
- [x] Make sure all tests pass
- [ ] Update documentation
- [ ] ~Check RBAC rights for Kubernetes / OpenShift roles~
- [ ] ~Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally~
- [ ] ~Reference relevant issue(s) and close them after merging~

Note: I have not yet tested this on master.

### CODE REVIEW

File: .travis/setup-kubernetes.sh [hunk 1]
```diff
@@ -5,6 +5,18 @@ function install_kubectl {
...
+function install_registry {
+    sudo -E $TEST_CLUSTER addons enable registry
+    # Wait for the registry pod to be ready
+
+    until $(kubectl --namespace=kube-system get po $(kubectl get po -n kube-system | grep 'registry-[a-z0-9]' | awk '{print $1;}') --no-headers | grep -Eq '([1-9]+[0-9]*)/\1'); do
+        printf '.';
+        sleep 5;
+    done
+
+    kubectl port-forward --namespace kube-system $POD 5000:5000 &
```
Comment (ppatierno) [inline .travis/setup-kubernetes.sh]: where the `$POD` var is set ?

Comment (tombentley) [reply .travis/setup-kubernetes.sh]: I was originally setting it like `POD=$(kubectl get po -n kube-system | grep 'registry-[a-z0-9]' | awk '{print $1;})`, which I was executing *before* the loop. The problem is that, initially when minikube starts it doesn't even know what pod resources are, so it went wrong before the loop was even entered. 

See the interesting discussion about why kubectl doesn't handle readiness already: https://github.com/kubernetes/kubernetes/issues/1899

**Comment by tombentley** [comment]
> Or how is thaat dealt with?

TBH, I have no idea. The registry plugin doesn't seem to have any documentation. It *had* documentation, but that got [removed](https://github.com/kubernetes/minikube/issues/2520), because apparently the YAML resources for it are documentation enough. So I don't know whether/how I need to take special steps to proxy the registry into the namespaces we use for testing, or whether than happens automagically.

> Also, will there be any issues with the secure / insecure registries

From what I've read `localhost:5000` is somehow special and means you don't need to mess around with certificates.

File: .travis/setup-kubernetes.sh (hunk 1 above)
Comment (ppatierno) [reply .travis/setup-kubernetes.sh]: I'm really dumb but I still don't understand from where the `$POD` var comes from or where it's set :-)

Comment (tombentley) [reply .travis/setup-kubernetes.sh]: @ppatierno I updated the PR, so there is no $POD anymore

**Comment by scholzj** [comment]
> TBH, I have no idea. The registry plugin doesn't seem to have any documentation. It had documentation, but that got removed, because apparently the YAML resources for it are documentation enough. So I don't know whether/how I need to take special steps to proxy the registry into the namespaces we use for testing, or whether than happens automagically.

So my guess would be that you will need to prefix all images with `localhost:5000`. Such as `image: localhost:8000/strimzi/cluster-controller:latest`. But I'm just guessing - it is not like I know this for sure.

**Comment by tombentley** [comment]
Let's not do it like this. Life's too short.

**Pull request closed without merging**